demisto_sdk/commands/validate/tests/BC_validators_test.py

```diff
@@ -1026,54 +1031,118 @@ def test_IsContextPathChangedValidator_remove_command():
+                create_modeling_rule_object(
+                    paths=["fromversion", "toversion"],
+                    values=["8.10.0", DEFAULT_CONTENT_ITEM_TO_VERSION],
+                )
             ],
-            id="Case 2: script - toversion changed",
+            0,
+            id="Case 4: Modeling Rule - toversion changed to 6.9.0 and the new fromversion is 6.10.0 which is a valid case",
         ),
     ],
 )
 def test_IsValidToversionOnModifiedValidator_obtain_invalid_content_items(
-    content_items, old_content_items
+    mocker, content_items, old_content_items, new_items, errors
 ):
     """
     Given:
-        - Case 1: two content item of type 'Integration', one with modified `toversion`.
-        - Case 2: two content item of type 'Script', one with modified `toversion`.
+        - Case 1: two content item of type 'Integration', `toversion` changed to 6.0.0 and the new `fromversion` is 6.10.0.
+        - Case 2: two content item of type 'ModelingRule', `toversion` changed to 6.0.0 and the new `fromversion` is 6.0.0.
+        - Case 3: two content items of type 'ModelingRule', toversion changed to 6.9.0 and no new item to replace it was found
+        - Case 4: two content items of type 'ModelingRule', toversion changed to 6.9.0 and the new fromversion is 6.10.0 which is a valid case
     When:
         - Calling the `IsValidToversionOnModifiedValidator` validator.
     Then:
-        - The obtain_invalid_content_items function will catch the change in `toversion` and will fail the validation only on the relevant content_item.
+        - Case 1: The obtain_invalid_content_items function will catch the change in `toversion` and will fail the validation since they are more than one release apart.
+        - Case 2: The obtain_invalid_content_items function will catch the change in `toversion` and will fail the validation since two items with the same id ,cannot exist in the same version.
+        - Case 3: The obtain_invalid_content_items function will catch the change in `toversion` and will fail the validation since no new item to replace it was found.
+        - Case 4: Valid case.
     """
     create_old_file_pointers(content_items, old_content_items)
+    for item in content_items:
+        item.git_status = GitStatuses.MODIFIED
+    for item in new_items:
+        item.git_status = GitStatuses.ADDED
+    content_items.extend(new_items)
+
     result = IsValidToversionOnModifiedValidator().obtain_invalid_content_items(
         content_items
     )
 
-    assert (
-        len(result) == 1
-        and result[0].message
-        == "Changing the maximal supported version field `toversion` is not allowed. Please undo, or request a force merge."
-    )
+    assert len(result) == errors
+    if result:
+        for res in result:
+            assert (
+                "Changing the maximal supported version field `toversion` is not allowed"
+                in res.message
+            )
```

Comment:
The point of the assertion is to assert the variables that are formatted into the message to ensure the errors were thrown for the right items.
Please assert the whole message including the object_id.